client/dashboard/sites/logs/dataviews/index.tsx

```diff
@@ -230,14 +246,58 @@ function SiteLogsDataViews( {
 		</>
 	);
 
+	const logs = logType === LogType.PHP ? phpLogs : serverLogs;
+
+	const infiniteScrollHandler = useCallback( () => {
+		if ( hasNextPage && ! isFetchingNextPage ) {
+			fetchNextPage();
+		}
+	}, [ hasNextPage, isFetchingNextPage, fetchNextPage ] );
+
+	useEffect( () => {
+		if ( ! dataviewsRef.current ) {
+			return;
+		}
+
+		const el = dataviewsRef.current;
+
+		const handleScroll = throttle( () => {
+			const scrollTop = el.scrollTop;
+			const scrollHeight = el.scrollHeight;
+			const clientHeight = el.clientHeight;
+
+			setShowScrollTop( scrollTop > clientHeight * 2 );
+
+			if ( scrollTop + clientHeight >= scrollHeight - 100 ) {
+				infiniteScrollHandler();
+			}
+		}, 100 );
+
+		el.addEventListener( 'scroll', handleScroll );
+		return () => el.removeEventListener( 'scroll', handleScroll );
+	}, [ infiniteScrollHandler ] );
+
+	const paginationInfo = {
+		totalItems: logs.length,
+		totalPages: 1,
+	};
+
+	useEffect( () => {
+		setView( ( currentView ) => ( {
+			...currentView,
+			perPage: Math.max( logs.length, currentView.perPage ?? DEFAULT_PER_PAGE ),
+		} ) );
+	}, [ logs.length, setView ] );
+
 	return (
 		<>
 			{ logType === LogType.PHP ? (
 				<DataViews< PHPLog >
-					data={ logs as PHPLog[] }
-					isLoading={ isFetching }
+					data={ phpLogs }
+					isLoading={ isLoading }
```

Comment:
Looks like we should keep using `isFetching` as the loading state didn't work correctly.